openeo/extra/job_management.py

```diff
@@ -1,5 +1,5 @@
 import contextlib
-import datetime
+from datetime import datetime, timedelta, timezone
```

Comment:
Fully agree

I made this revision since I was struggling with datetime for this exact reason in the unit tests. I wanted to mock the datetime.datetime module without mocking the datetime.timedelta module, and this was only possible when importing it as is. 

Maybe good to revisit this comment when you had had time to look at the tests as well.

Reply:
regarding mocking datetime in tests, we started using the "time_machine" test utility, which takes care of all that hard-to-get-right date/time mocking stuff.
e.g. see https://github.com/Open-EO/openeo-python-client/blob/f7ee55223464bf67e45291b9df773d3749f4150f/tests/test_util.py#L297-L305

Reply:
by moving to rcf3339, I was able to avoid the datetime ambiguity